keras_nlp/applications/bert.py

```diff
@@ -0,0 +1,372 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Bert model and layer implementations."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+import keras_nlp.layers
+
+# isort: off
+# TODO(bischof): decide what to export or whether we are using these decorators
+from tensorflow.python.util.tf_export import keras_export
+
+CLS_INDEX = 0
+TOKEN_EMBEDDING_LAYER_NAME = "token_embedding"
+
+
+def BertEncoder(
+    vocab_size,
+    num_layers=12,
+    hidden_size=768,
+    dropout=0.1,
+    num_attention_heads=12,
+    inner_size=3072,
+    inner_activation="gelu",
+    initializer_range=0.02,
+    max_sequence_length=512,
+    type_vocab_size=2,
```

Comment:
Should we take a weights argument here too? Just have no presets on this class?

Maybe easier to answer once we have figure out how to deliver preprocessing.

Reply:
If we're using the `Model` subclass then the `load_weights()` functionality is already available. I think we only need explicit handling for the fully baked models like `BertBase` below.